build.gradle

```diff
@@ -648,6 +648,8 @@ project('shared:controller-api') {
         compile "io.grpc:grpc-protobuf:" + grpcVersion
         compile "io.grpc:grpc-stub:" + grpcVersion
         compile group: 'io.netty', name: 'netty-tcnative-boringssl-static', version: nettyBoringSSLVersion
+        // override grpc's transitive protobuf version with allprojects' force-upgraded version
+        compile group: 'com.google.protobuf', name: 'protobuf-java', version: protobufProtocVersion
```

Comment:
From the dependency list for Pravega-client we can observe that the client (via `shared:controller-api` ) already depends on `protobuf-java` with the correct version i.e. (protobufProtocVersion=3.13.0)

Dependency Snip :
```
|    +--- io.grpc:grpc-protobuf:1.24.2
|    |    +--- io.grpc:grpc-api:1.24.2 (*)
|    |    +--- com.google.protobuf:protobuf-java:3.9.0 -> 3.13.0
```
The above snip indicates the project already depends on 3.13.0 version for `protobuf-java` dependency.  This dependency is added as part of the `protobuf.gradle` which auto added when apply plugin : `com.google.protobuf` is used .
Hence I think this change here is not required...

Complete depencdency output : 
[Client-deps.txt](https://github.com/pravega/pravega/files/6080971/Client-deps.txt)

Reply:
@shrids -- The problem is that that override does not become a part of the jar's Maven pom xml generated by Gradle, so when the jar is used external to the Gradle build, the user will get the wrong version of the dependency. Adding this line ensures that the jar will also require the version of protobuf-java that it was build with. You can check this by bringing the jar into an external project.

Reply:
Thanks for the clarification. 
Could you add this dependency inside the protobuf.gradle file so that any project that does apply plugin : com.google.protobuf overrides the version of `protobuf-java` ?

Reply:
@shrids Interesting idea! I'll check if that works.